app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx

```diff
@@ -348,6 +349,45 @@ export default class SampleViewV2 extends React.Component {
     return parsedValue;
   };
 
+  parseSpeciesMetricDecimalPlaces = species => {
+    let parsedSpeciesMetrics = Object.assign({}, species);
+    let metricDecimalPlaces = {
+      agg_score: 0,
+      z_score: 1,
+      rpm: 1,
+      count: 0,
+      contigs: 0,
+      contig_r: 0,
+      percent_identity: 1,
+      alignment_length: 1,
+      e_value: 1,
+    };
+
+    Object.keys(species).forEach(outerMetric => {
+      if (outerMetric in metricDecimalPlaces) {
+        parsedSpeciesMetrics[outerMetric] = parseFloat(
+          species[outerMetric]
+        ).toFixed(metricDecimalPlaces[outerMetric]);
+      } else if (outerMetric === "nt") {
```

Comment:
If you use Object.entries as above, this branch can be treated like the bottom one. Shorter code :)